File: src/site/wenlong/dimens/languages/English.java
package site.wenlong.dimens.languages;

/**
 * 语言国际化-英文
 *
 * @author : 郭文龙
 * @date : 2019/4/30  23:30
 */
public class English implements Text {
    private String titleOne = "Setting Options";
    private String cover = "Overwrite existing files";
    private String decimal = "Keep the decimal point";
    private String minWidth = "Set the minimum width of the selected dimens.xml(Default 360dp)";
    private String folder = "Set the qualifier name (default sw) (Note: no matter how you name it will not change the rules for file generation)";

    private String bit = "bit";
    private String dp = "dp";
    private String more = "more";

    private String titleTwo = "Generation Mode";

    private String single = "Single";
    private String multiple = "Multiple";

    private String advancedOption = "advanced option";
    private String genarate = "Genarate dimens";

    //提示语
    private String tipsErrorFile = "The file you selected is not dimens.xml, please re-select";
    private String layoutTipsErrorFile = "The file you selected is not layout folder or an xml file, please re-select";
    private String tipsErrorNumber = "Please enter the correct number";

    private static English instance;

    public static synchronized English getInstance() {
        if (instance == null) {
            instance = new English();
        }
        return instance;
    }

    @Override
    public String getTitleOne() {
        return titleOne;
    }

    @Override
    public String getCover() {
        return cover;
    }

    @Override
    public String getDecimal() {
        return decimal;
    }

    @Override
    public String getMinWidth() {
        return minWidth;
    }

    @Override
    public String getFolder() {
        return folder;
    }

    @Override
    public String getBit() {
        return bit;
    }

    @Override
    public String getDp() {
        return dp;
    }

    @Override
    public String getMore() {
        return more;
    }

    @Override
    public String getTitleTwo() {
        return titleTwo;
    }

    @Override
    public String getSingle() {
        return single;
    }

    @Override
    public String getMultiple() {
        return multiple;
    }

    @Override
    public String getAdvancedOption() {
        return advancedOption;
    }

    public String getTipsErrorFile() {
        return tipsErrorFile;
    }

    @Override
    public String getLayoutTipsErrorFile() {
        return layoutTipsErrorFile;
    }

    @Override
    public String getGenarate() {
        return genarate;
    }

    @Override
    public String getTipsErrorNumber() {
        return tipsErrorNumber;
    }

    @Override
    public String getTipsCreateFileFailed() {
        return "File does not exist or folder creation failed";
    }

    @Override
    public String getTipsCreateFileError() {
        return "Generate xml file or folder exception, please submit a question to github, thanks";
    }

    @Override
    public String getTipsGenerateSuccess() {
        return "Generated successfully";
    }

    @Override
    public String getTipsConvertSuccess() {
        return "Converted successfully";
    }

    @Override
    public String getTipsDimensExists() {
        return "The dimens.xml file of the %s folder already exists.\n Please check the Dimens Generating Tools menu to overwrite the source file \n or delete and regenerate after backup.";
    }

    @Override
    public String getTipsInputEmpty() {
        return "Do not allow input to be empty";
    }

    @Override
    public String getTipsInputZero() {
        return "Do not allow input numbers to be 0";
    }

    @Override
    public String getTipsConvertTitle() {
        return "Convert Option";
    }

    @Override
    public String getTipsConvertTip1() {
        return "Tips:need a little time\n      2.px will not convert";
    }

    @Override
    public String getTipsConvertDP() {
        return "dp conversion prefix";
    }

    @Override
    public String getTipsConvertSP() {
        return "sp conversion prefix";
    }

    @Override
    public String getTipsConvertEP1() {
        return "(For example: length_ ,generation is length_1)";
    }

    @Override
    public String getTipsConvertEP2() {
        return "(For example: font_ ,generation is font_1)";
    }

    @Override
    public String getTipsConvertButton() {
        return "Convert";
    }
}

File: src/main/kotlin/site/wenlong/dimens/languages/English.kt
package site.wenlong.dimens.languages

/**
 * 英语
 *
 * @author : 郭文龙
 * @Email  : guowenlong20000@sina.com
 * @date   : 2023/5/6  23:14
 */
object English : Text {
    override val titleOne = "Generate Options"

    override val cover = "Override existing files"

    override val decimal = "Keep the decimal point"

    override val minWidth = "Set the minimum width of the selected dimens.xml(Default 360dp)"

    override val folder = "Set qualifier name (default is \"sw\") (note: naming does not affect file generation rules)"

    override val bit = "bit"

    override val more = "more"

    override val titleTwo = "Generation Mode"

    override val single = "Single"

    override val multi = "Multiple"

    override val advancedOption = "advanced option"

    override val genarate = "Generate dimens"

    override val tipsErrorFile = "The file you selected is not dimens.xml, please re-select"

    override val layoutTipsErrorFile =
        "The file you selected is not layout folder or an xml file, please re-select"

    override val tipsErrorNumber = "Please enter the correct number"

    override val tipsCreateFileFailed: String = "File does not exist or folder creation failed"

    override val tipsCreateFileError: String =
        "Generate xml file or folder exception, please submit a question to github, thanks";

    override val tipsGenerateSuccess: String = "Generated successfully"

    override val tipsGenerateFail: String = "Generated fail"

    override val tipsConvertSuccess: String = "Converted successfully"

    override val tipsConvertFail: String = "Converter fail"

    override val tipsDimensExists: String =
        "The dimens.xml file of the %s folder already exists.\n Please check the Dimens Generating Tools menu to overwrite the source file \n or delete and regenerate after backup."

    override val tipsInputEmpty: String = "Do not allow input to be empty"

    override val tipsInputZero: String = "Do not allow input numbers to be 0"

    override val tipsConvertTitle: String = "Convert Option"

    override val tipsConvertTip1: String = "Tips:need a little time and px will not be convert"

    override val tipsConvertDP: String = "dp conversion prefix"

    override val tipsConvertSP: String = "sp conversion prefix"

    override val tipsConvertEP1: String = "(For example: length_ ,generation is length_1)"

    override val tipsConvertEP2: String = "(For example: font_ ,generation is font_1)"

    override val tipsConvertButton: String = "Convert"

    override val language: String = "Language"

    override val converting: String = "Converting..."

    override val exceptionOfFileError: String = "File Error"
}